File: Pod/Generator/project/project.py
from mod_pbxproj import XcodeProject
import json
import sys

pbxproj_file_path = sys.argv[1]
json_file_path = sys.argv[2]
project = XcodeProject.Load(pbxproj_file_path)
#project = XcodeProject.Load('../../../DKNightVersion.xcodeproj/project.pbxproj')

project.remove_group_by_name('UIKit', True)

project.save()

main_group = project.get_or_create_group('DKNightVersion')
pod_group = project.get_or_create_group('Pod', None, main_group)
class_group = project.get_or_create_group('Classes', None, pod_group)
uikit_group = project.get_or_create_group('UIKit', None, class_group)

groups = json.load(open(json_file_path))

for group, files in groups.iteritems():
    new_group = project.get_or_create_group(group, None, uikit_group)
    for f in files:
        project.add_file(f, new_group)

project.save()

File: project.rb
require 'xcodeproj'
require 'json'

def extract_children_by_name(group, name)
    group.children.each do |child|
        if child.path == name
            puts child.inspect
            return child
        end
    end
end
path = 'DKNightVersion/Pod/Classes/UIKit/'
j = JSON.parse File.read('./project.json')
project = Xcodeproj::Project.open('DKNightVersion.xcodeproj')

#uikit_group = project.main_group.find_subpath(path, true)

j.each do |group, files|
    new_group = project.main_group.find_subpath(File.join(path, group), true)
    files.each do |f|
        puts f
        new_group.new_file(f)
    end
end

#uikit_group.new_reference('UIKit/UIButton/UIButton+NightVersion.h')

project.save


